### ISSUE TITLE
Question about the input and output size of the VRN

### ISSUE THREAD

**Issue by pengjingg** [Ryo-Ito/brain_segmentation#10]
Hello Ryo,
I have two questions about the structure of VoxResNet

First, why the input size is [80, 80, 80] and the output size is [60, 60, 60] ?
The paper didn't mention about the input and output size of the network,  is it ok to set the output size equals to input size?

Second, for some reason, my results are blobby and weird. I guess it's because the split and stitch operation of the image, but I cannot find a way to figure it out. I'm using IBSR dataset (training set: 01-05, testing set: 11-15) and the image size after preprocessing is 256×256×256

![screenshot from 2017-11-22 16-08-25](https://user-images.githubusercontent.com/25976861/33118317-66619476-cfa6-11e7-98cc-79dde6209044.png)

Thanks a lot !

**Comment by pengjingg** [comment]
All right, the problem has been solved by myself. The blobby result indicates the incorrect prediction of the backgroud. In the original code by Ryo, I find that only regions with a positive label in the input data have been trained. Background has never been trained, which causes the chaotic output of background area. You can find the relative code in `utils.py`,  `load_sample()`:

```
for image_file, label_file in zip(image_files, label_files):
        image = load_nifti(image_file)
        label = load_nifti(label_file).astype(np.int32)
        mask = np.int32(label > 0)
        slices = [slice(len_ / 2, -len_ / 2) for len_ in input_shape]
        mask[slices] *= 2
        indices = np.where(mask > 1.5)
        i = np.random.choice(len(indices[0]))
```
where `mask` limited the choices of input patch center to be in regions with label>0
Here is my segmentation result after the correction of the code:
(left: segmentation; middle: ground truth; right: error)

![screenshot from 2017-12-01 11-02-03](https://user-images.githubusercontent.com/25976861/33466146-1c57c354-d687-11e7-99a2-9a29bba4724d.png)

**Comment by Mathijssch** [comment]
@pengjingg Did you train with the output size equal to the input size when you encountered this problem?

**Comment by pengjingg** [comment]
@Mathijssch Of course

**Comment by Mathijssch** [comment]
Another question, @pengjingg , was your solution just to disable the mask by setting it to 1 everywhere, or did you apply a more sophisticated strategy to still limit the probability of selecting a pure background patch?

**Comment by pengjingg** [comment]
@Mathijssch No. I didn't use any strategy to prevent the network from choosing pure background as input. Because in my experience, if backgrounds are not trained and learned well, the output of background areas will be messy.
